scripts/validateLinks.js

```diff
@@ -0,0 +1,161 @@
+        console.log(`   📄 Pages checked:  ${pagesChecked.size.toLocaleString()}`)
+        console.log(`   📝 Total checked:  ${results.links.length.toLocaleString()}`)
+
+        // Check for issues
+        if (validLinks === 0 && results.links.length > 0) {
+            console.log('\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━')
+            console.log('❌ VALIDATION FAILED')
+            console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━')
+            console.log(`\n⚠️  Found ${results.links.length} links but 0 valid.`)
+            console.log('\n💡 Possible causes:')
+            console.log('   • Server not running at specified URL')
+            console.log('   • All links skipped by skip patterns')
+            console.log('   • Network connectivity issues\n')
+            process.exit(1)
+        }
+
+        if (brokenLinks.length > 0) {
+            console.log('\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━')
+            console.log('🔴 BROKEN LINKS DETAILS')
+            console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━')
+
+            // Group by parent page
+            const byPage = {}
+            brokenLinks.forEach(link => {
+                if (!byPage[link.parent]) byPage[link.parent] = []
+                byPage[link.parent].push(link)
+            })
+
+            Object.entries(byPage).forEach(([page, links], i) => {
+                console.log(`\n📄 Page ${i + 1}: ${page}`)
+                links.forEach((link, j) => {
+                    console.log(`\n   ${j + 1}. ❌ ${link.url}`)
+                })
+            })
+
+            // Save report
+            const timestamp = new Date().toISOString().replace(/:/g, '-').split('.')[0]
+            const reportPath = path.resolve(__dirname, '..', `broken-links-report-${timestamp}.json`)
+
+            fs.writeFileSync(reportPath, JSON.stringify({
+                date: new Date().toISOString(),
+                url,
+                summary: {
+                    total: results.links.length,
+                    valid: validLinks,
+                    broken: brokenLinks.length,
+                    skipped: skippedLinks,
+                    pagesChecked: Array.from(pagesChecked)
+                },
+                brokenLinks: brokenLinks.map(link => ({
+                    url: link.url,
+                    parent: link.parent,
+                    status: link.statusCode
+                }))
+            }, null, 2))
+
+            console.log(`\n💾 Report saved to: ${reportPath}\n`)
+            process.exit(1)
+        } else {
+            console.log('\n✅ SUCCESS! No broken links found.\n')
+        }
+    } catch (error) {
+        console.error('Error running link checker:', error.message)
+        process.exit(1)
+    }
+}
+
+// Parse args and run
+const args = process.argv.slice(2)
+const silent = args.includes('--silent')
+const url = args.filter(arg => arg !== '--silent')[0] || DEFAULT_URL
```

Comment:
If you'd like to familiarize yourself with a common CLI framework, try doing this with [commander](https://www.npmjs.com/package/commander). We use it in Pepr and a few other places.